[CC265x example][Building error]error: initializer element is not a constant expression [-Wpedantic]

Hi,

After changes introduced in #2361 the build process went further for this example but I'm still getting some errors. 



`In file included from /home/aron/devel/openthread/examples/../examples/platforms/cc2650/cc2650_ccfg.c:48:0:
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:389:42: error: initializer element is not a constant expression [-Wpedantic]
 #define DEFAULT_CCFG_MODE_CONF           ( \
                                          ^
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:480:5: note: in expansion of macro 'DEFAULT_CCFG_MODE_CONF'
     DEFAULT_CCFG_MODE_CONF          , // 0x50003FB4
     ^~~~~~~~~~~~~~~~~~~~~~
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:389:42: note: (near initialization for '__ccfg.CCFG_MODE_CONF')
 #define DEFAULT_CCFG_MODE_CONF           ( \
                                          ^
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:480:5: note: in expansion of macro 'DEFAULT_CCFG_MODE_CONF'
     DEFAULT_CCFG_MODE_CONF          , // 0x50003FB4
     ^~~~~~~~~~~~~~~~~~~~~~
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:404:42: error: initializer element is not a constant expression [-Wpedantic]
 #define DEFAULT_CCFG_VOLT_LOAD_0         ( \
                                          ^
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:481:5: note: in expansion of macro 'DEFAULT_CCFG_VOLT_LOAD_0'
     DEFAULT_CCFG_VOLT_LOAD_0        , // 0x50003FB8
     ^~~~~~~~~~~~~~~~~~~~~~~~
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:404:42: note: (near initialization for '__ccfg.CCFG_VOLT_LOAD_0')
 #define DEFAULT_CCFG_VOLT_LOAD_0         ( \
                                          ^
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:481:5: note: in expansion of macro 'DEFAULT_CCFG_VOLT_LOAD_0'
     DEFAULT_CCFG_VOLT_LOAD_0        , // 0x50003FB8
     ^~~~~~~~~~~~~~~~~~~~~~~~
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:410:42: error: initializer element is not a constant expression [-Wpedantic]
 #define DEFAULT_CCFG_VOLT_LOAD_1         ( \
                                          ^
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:482:5: note: in expansion of macro 'DEFAULT_CCFG_VOLT_LOAD_1'
     DEFAULT_CCFG_VOLT_LOAD_1        , // 0x50003FBC
     ^~~~~~~~~~~~~~~~~~~~~~~~
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:410:42: note: (near initialization for '__ccfg.CCFG_VOLT_LOAD_1')
 #define DEFAULT_CCFG_VOLT_LOAD_1         ( \
                                          ^
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:482:5: note: in expansion of macro 'DEFAULT_CCFG_VOLT_LOAD_1'
     DEFAULT_CCFG_VOLT_LOAD_1        , // 0x50003FBC
     ^~~~~~~~~~~~~~~~~~~~~~~~
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:416:42: error: initializer element is not a constant expression [-Wpedantic]
 #define DEFAULT_CCFG_RTC_OFFSET          ( \
                                          ^
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:483:5: note: in expansion of macro 'DEFAULT_CCFG_RTC_OFFSET'
     DEFAULT_CCFG_RTC_OFFSET         , // 0x50003FC0
     ^~~~~~~~~~~~~~~~~~~~~~~
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:416:42: note: (near initialization for '__ccfg.CCFG_RTC_OFFSET')
 #define DEFAULT_CCFG_RTC_OFFSET          ( \
                                          ^
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:483:5: note: in expansion of macro 'DEFAULT_CCFG_RTC_OFFSET'
     DEFAULT_CCFG_RTC_OFFSET         , // 0x50003FC0
     ^~~~~~~~~~~~~~~~~~~~~~~
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:421:42: error: initializer element is not a constant expression [-Wpedantic]
 #define DEFAULT_CCFG_FREQ_OFFSET         ( \
                                          ^
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:484:5: note: in expansion of macro 'DEFAULT_CCFG_FREQ_OFFSET'
     DEFAULT_CCFG_FREQ_OFFSET        , // 0x50003FC4
     ^~~~~~~~~~~~~~~~~~~~~~~~
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:421:42: note: (near initialization for '__ccfg.CCFG_FREQ_OFFSET')
 #define DEFAULT_CCFG_FREQ_OFFSET         ( \
                                          ^
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:484:5: note: in expansion of macro 'DEFAULT_CCFG_FREQ_OFFSET'
     DEFAULT_CCFG_FREQ_OFFSET        , // 0x50003FC4
     ^~~~~~~~~~~~~~~~~~~~~~~~
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:431:42: error: initializer element is not a constant expression [-Wpedantic]
 #define DEFAULT_CCFG_BL_CONFIG           ( \
                                          ^
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:489:5: note: in expansion of macro 'DEFAULT_CCFG_BL_CONFIG'
     DEFAULT_CCFG_BL_CONFIG          , // 0x50003FD8
     ^~~~~~~~~~~~~~~~~~~~~~
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:431:42: note: (near initialization for '__ccfg.CCFG_BL_CONFIG')
 #define DEFAULT_CCFG_BL_CONFIG           ( \
                                          ^
/home/aron/devel/openthread/examples/../third_party/ti/devices/cc26x0/startup_files/ccfg.c:489:5: note: in expansion of macro 'DEFAULT_CCFG_BL_CONFIG'
     DEFAULT_CCFG_BL_CONFIG          , // 0x50003FD8
     ^~~~~~~~~~~~~~~~~~~~~~
Makefile:771: recipe for target 'libopenthread_cc2650_a-cc2650_ccfg.o' failed  `

--- a/.travis/script.sh
+++ b/.travis/script.sh
@@ -243,21 +243,21 @@ set -x
     arm-none-eabi-size  output/nrf52840/bin/ot-ncp-ftd || die
     arm-none-eabi-size  output/nrf52840/bin/ot-ncp-mtd || die
 
-    # git checkout -- . || die
-    # git clean -xfd || die
-    # ./bootstrap || die
-    # make -f examples/Makefile-cc2650 || die
-    # arm-none-eabi-size  output/cc2650/bin/ot-cli-mtd || die
-    # arm-none-eabi-size  output/cc2650/bin/ot-ncp-mtd || die
+    git checkout -- . || die
+    git clean -xfd || die
+    ./bootstrap || die
+    make -f examples/Makefile-cc2650 || die
+    arm-none-eabi-size  output/cc2650/bin/ot-cli-mtd || die
+    arm-none-eabi-size  output/cc2650/bin/ot-ncp-mtd || die
 
-    # git checkout -- . || die
-    # git clean -xfd || die
-    # ./bootstrap || die
-    # COMMISSIONER=1 JOINER=1 DHCP6_CLIENT=1 DHCP6_SERVER=1 DNS_CLIENT=1 make -f examples/Makefile-cc2652 || die
-    # arm-none-eabi-size  output/cc2652/bin/ot-cli-ftd || die
-    # arm-none-eabi-size  output/cc2652/bin/ot-cli-mtd || die
-    # arm-none-eabi-size  output/cc2652/bin/ot-ncp-ftd || die
-    # arm-none-eabi-size  output/cc2652/bin/ot-ncp-mtd || die
+    git checkout -- . || die
+    git clean -xfd || die
+    ./bootstrap || die
+    COMMISSIONER=1 JOINER=1 DHCP6_CLIENT=1 DHCP6_SERVER=1 DNS_CLIENT=1 make -f examples/Makefile-cc2652 || die
+    arm-none-eabi-size  output/cc2652/bin/ot-cli-ftd || die
+    arm-none-eabi-size  output/cc2652/bin/ot-cli-mtd || die
+    arm-none-eabi-size  output/cc2652/bin/ot-ncp-ftd || die
+    arm-none-eabi-size  output/cc2652/bin/ot-ncp-mtd || die
 
     git checkout -- . || die
     git clean -xfd || die

--- a/examples/platforms/cc2650/Makefile.am
+++ b/examples/platforms/cc2650/Makefile.am
@@ -30,6 +30,11 @@ include $(abs_top_nlbuild_autotools_dir)/automake/pre.am
 
 lib_LIBRARIES                                         = libopenthread-cc2650.a
 
+# Do not enable -pedantic-errors for cc26xx driverlib
+override CFLAGS                                      := $(filter-out -pedantic-errors,$(CFLAGS))
+override CXXFLAGS                                    := $(filter-out -pedantic-errors,$(CXXFLAGS))
+
+
 libopenthread_cc2650_a_CPPFLAGS                       = \
     -I$(top_srcdir)/include                             \
     -I$(top_srcdir)/src                                 \

--- a/examples/platforms/cc2652/Makefile.am
+++ b/examples/platforms/cc2652/Makefile.am
@@ -30,6 +30,10 @@ include $(abs_top_nlbuild_autotools_dir)/automake/pre.am
 
 lib_LIBRARIES                                         = libopenthread-cc2652.a
 
+# Do not enable -pedantic-errors for cc26xx driverlib
+override CFLAGS                                      := $(filter-out -pedantic-errors,$(CFLAGS))
+override CXXFLAGS                                    := $(filter-out -pedantic-errors,$(CXXFLAGS))
+
 libopenthread_cc2652_a_CPPFLAGS                       = \
     -I$(top_srcdir)/include                             \
     -I$(top_srcdir)/src                                 \

